### PR TITLE
Dependencies: Use `latest` version of `vue-tsc` & sync versions of `angular`

### PR DESCRIPTION
**Pull request by ndelangen**
some volar update caused vue to start vomitting ts-errors

It seems that when a file file filename `Button.vue` is used, it will define a component `button`.
I'm not sure of this is a bug or a feature.

But if `Button.vue` uses `<button>` (which would obviously refer to `HTMLButtonElement`), volar gets "confused" and assumes thus button is the button being defined.

But of course the API (props, events) of those 2 are different, thus it complains.

I renamed `Button.vue` to `MyButton.vue`, this fixes it.

I'm not really happy with this though.

### CODE REVIEW

File: code/yarn.lock [hunk 1]
```diff
@@ -9562,13 +9574,13 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@vue/typescript@npm:1.7.8":
-  version: 1.7.8
-  resolution: "@vue/typescript@npm:1.7.8"
+"@vue/typescript@npm:1.7.11":
```
Comment (ndelangen) [inline code/yarn.lock:9577]: @kasperpeulen

File: code/yarn.lock [hunk 2]
```diff
@@ -9442,12 +9454,12 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@vue/language-core@npm:1.7.8":
-  version: 1.7.8
-  resolution: "@vue/language-core@npm:1.7.8"
+"@vue/language-core@npm:1.7.11":
```
Comment (ndelangen) [inline code/yarn.lock:9457]: @kasperpeulen

File: code/yarn.lock [hunk 3]
```diff
@@ -9275,30 +9287,30 @@ __metadata:
...
     muggle-string: ^0.3.1
-  checksum: 94e0128e161f1709c3b1996dde39d891074b40c69cb53e017a01d337c7ad9089423ffdaedfdf01099ca6f793211233ef9b2230d241970afcf1a587728f970720
+  checksum: d04fd02c1d1548b1d65893003a350cf3686f8069cc6f281bd8e5d7e33dc45b8b72cd35a9243cebaa3afbcd0b5615ffc5dc6dacc6e55fd9e657a6aeefec2da15e
   languageName: node
   linkType: hard
 
-"@volar/typescript@npm:1.6.9":
-  version: 1.6.9
-  resolution: "@volar/typescript@npm:1.6.9"
+"@volar/typescript@npm:1.7.4":
```
Comment (ndelangen) [inline code/yarn.lock:9308]: @kasperpeulen

File: code/yarn.lock [hunk 4]
```diff
@@ -9275,30 +9287,30 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@volar/language-core@npm:1.6.9":
-  version: 1.6.9
-  resolution: "@volar/language-core@npm:1.6.9"
+"@volar/language-core@npm:1.7.4":
```
Comment (ndelangen) [inline code/yarn.lock:9290]: @kasperpeulen

**Comment by ndelangen** [comment]
This seems related:
https://github.com/storybookjs/storybook/issues/22853#issuecomment-1584105589

Seems `@volar/typescript` is on npm on the latest tag with an actual prerelease?!?

I'll downgrade back to what they deem to be `latest` and lock it.
Why aren't they simply publishing prereleases as a separate tag?

**Comment by ndelangen** [comment]
@Sidnioulz here's a PR of us switching us to the "latest" release according to their CHANGELOG.

Do you expect this will cause issues for storybook users?
Or will it actually fix issues?

**Review by kasperpeulen — APPROVED** [review]
I think this is great, let's in an other PR also put `vue-component-type-helpers` to 'latest' to fix the issues of @Sidnioulz

**Comment by Sidnioulz** [comment]
@ndelangen sorry, mail goes to my personal inbox and it takes time to process :(

I don't expect any change or improvement with vue-component-type-helpers. In my limited Vue experience, the issue you describe is bound to happen because of single-component naming.

It's a convention in Vue that component names must be multi-word. The Vue maintainers provide an ESLint rule to enforce that: https://eslint.vuejs.org/rules/multi-word-component-names.html.

Without enforcing the rule, we've seen other linters in the past complain about `Button.vue` and whatnot not respecting the APIs of their HTML standard counterparts. The expectation is so strong in the Vue community that we just accept it and use a prefix on our Vue component names.

Anyone struggling with this and wanting to keep their file named `Button.vue` can define the name explicitly:

```ts
// Vue 3 composition API
defineOptions({
  name: 'MyButton',
})

// Vue 3 options API, never used it but I guess it's that
export default {
  name: 'MyButton',
  ...
}

// Vue 2 with property decorators / class components
@Component
export default class MyButton extends Vue { ... }
```

**Comment by ndelangen** [comment]
Good to know @Sidnioulz, to the problem I faced, will come back to haunt us later.. And might be actually hunting users of storybook currently?

We should perhaps setup this mentioned eslint rule on the vue part of our codebase.

Any help would be appreciated @Sidnioulz if you have the time 🙏 
I know @chakAs3 and @kasperpeulen are the resident vue-experts, they might be keen to know this as well: https://github.com/storybookjs/storybook/pull/23011#issuecomment-1588618885

**Comment by chakAs3** [comment]
Hi @ndelangen   indeed Vue uses and recommends using PascalCase names because they are valid JavaScript identifiers so easy to register and import, and easy for IDE to auto-complete. <PascalCase />  means it Vue component and not a native or custom component ( web-component ).

However, Vue can resolve to kebab-case as well so both <MyComponent/> and <my-component/> are valid. 
the take from this is we should avoid Naming our Component name that can resolve to a native Element ex: <Button/>  => <button>, <Table> => <table>, the solution that comes to mind is PascalCase, which resolves to <kebab-case/> and as we know there is no native element in kebab case

**Comment by chakAs3** [comment]
@Sidnioulz  yeah right it is really upto the developers if they want to pay it save and clean, despite the tools and rules that help have a saint codebase you can always screw it up believe me.

> but passed via the `components` option, then the name you use to import them must match the `name` option (can't explain why any more, but that caused bugs somewhere)

Yes, but to be honest I don't do much code using options to define. the component name unless you really want to be explicit about the component name, this most for Ui libraries.
For me, in certain cases, the component does not play a big role in my code, within my work on Volar component meta i was doing something like this for tests.
image

I could a bit more specific and just change the import 
```ts
import type { StoryObj, Meta } from '@storybook/vue3';

import SlotsComponent from './template-slots/component.vue';

const meta = {
  component: SlotsComponent,
  tags: ['autodocs'],
} satisfies Meta<typeof SlotsComponent>;
type Story = StoryObj<typeof meta>;
export default meta;

export const Default: Story = {
  args: {
    default: ({ num }) => `Default slot { num=${num} }`,
    named: ({ str }) => `Named slot { str=${str} }`,
    vbind: ({ num, str }) => `Named v-bind slot { num=${num}, str=${str} }`,
  },
};
```

**Pull request merged**